tools/cli/README.md

```diff
@@ -0,0 +1,71 @@
+## What
+Cadence CLI provide an command-line tool for users to perform various tasks on Cadence.  
+Users can perform operations like register, update and describe on domain;  
+also start workflow, show workflow history, signal workflow ... and many other tasks.    
+
+## How
+- Run `make bins`
+- You should see an executable `cadence-cli`
+
+## Quick Start
+Run `./cadence-cli` to view help message. There are some top level commands and global options.   
+Run `./cadence-cli domain` to view help message about operations on domain  
+Run `./cadence-cli workflow` to view help message about operations on workflow  
+
+*Note:* make sure you have cadence server running before using CLI 
+
+### Domain operation examples 
+- Register a new domain named "samples-domain":  
+```
+./cadence-cli --domain samples-domain domain register 
+# OR using short alias  
+./cadence-cli --do samples-domain domain re
+```   
+- View "samples-domain" details:   
+```
+./cadence-cli --domain samples-domain domain describe  
+```
+
+### Workflow operation examples
+
+- Start workflow: (for example [this workflow](https://github.com/samarabbas/cadence-samples/blob/master/cmd/samples/recipes/helloworld/helloworld_workflow.go))
+```
+./cadence-cli --domain samples-domain workflow start --tl helloWorldGroup --wt main.Workflow --et 60 --dt 10 -i '"vancexu"'
+
+# see more options for workflow start
+./cadence-cli --domain samples-domain workflow start -h
+
+# for workflow with multiple input, seperate each json with space/newline like
+./cadence-cli --domain samples-domain workflow start --tl helloWorldGroup --wt main.WorkflowWith3Args --et 60 --dt 10 -i '"your_input_string" 123 {"Name":"my-string", "Age":12345}'
+```
+
+- Run workflow and see the progress 
+```
+./cadence-cli --domain samples-domain workflow run --tl helloWorldGroup --wt main.Workflow --et 60 --dt 10 -i '"vancexu"'    
+``` 
+Note you need to start worker so that workflow can make progress.  
+
+- List open or closed workflow executions
+```
+./cadence-cli --domain samples-domain workflow list
+```
+
+- Show workflow history
+```
+./cadence-cli --domain samples-domain workflow show -w 3ea6b242-b23c-4279-bb13-f215661b4717 -r 866ae14c-88cf-4f1e-980f-571e031d71b0
+
+# if run_id is not provided, it will show the latest run history for that workflow_id
+./cadence-cli --domain samples-domain workflow show -w 3ea6b242-b23c-4279-bb13-f215661b4717
+```
+
+*Tips:*  
+to avoid repeated input global option **domain**, user can export domain-name in environment variable CADENCE_CLI_DOMAIN.
+(There are also other environment variables like this, see help message for details)
+```
+export CADENCE_CLI_DOMAIN=samples-domain
+
+# then you can just
+./cadence-cli domain desc
+./cadence-cli workflow list
+./cadence-cli workflow show -w greetings_38a9e481-a00c-433a-b1ff-b6f67536efd8
```

Comment:
Can we make -w optional after show? Just workflow show UUID?

Reply:
With the cli framework we currently use, it would be a little bit tricky to achieve this requirement. Because `show` commands also accepts other optional arguments. 
Instead, I propose to add new command like `showid <wid>` as to separate command which can only take wid and have no other arguments to achieve this. what do you think?

Reply:
If it is indeed that hard then showid is OK.

Reply:
Cool, added `showid <wid>` as a shortcut of `show -w <wid>`